### PR TITLE
Maze Layout System

### PR DESCRIPTION
**Pull request by IsaiahBlanks**
# Description
This PR adds a system for users to add their own custom maze layouts. These are represented in JSON by 2D arrays, and in the game as areas where mazes tiles do or do not generate. Also in this PR, WorldBuilder is renamed to GalaxyBuilder.

# Testing
Fly around until you find a smaller Maze. If you are lucky, it will by a SpiralMaze. You'll be able to tell by the spiral shape the maze forms (it is a bit too big to see at once, fly around the maze a bit to tell).

# Outstanding Work
This one involves loading in JSON so I wasn't sure how to write unit tests for it, but that still need to be done.

### CODE REVIEW

File: engine/src/main/java/org/destinationsol/game/maze/MazeLayoutBuilder.java [hunk 1]
```diff
@@ -51,6 +51,21 @@ public MazeLayout build() {
         return new MazeLayout(inners, holes, right, down);
     }
 
+    public MazeLayout build(boolean[][] innerPreset) {
```
Comment (IsaacLic) [inline engine/src/main/java/org/destinationsol/game/maze/MazeLayoutBuilder.java:54]: Could you add a bit of documentation to explain how this works?

File: engine/src/main/java/org/destinationsol/world/generators/SmallMazeGenerator.java [hunk 2]
```diff
@@ -15,14 +15,11 @@
...
 
-import org.slf4j.Logger;
-import org.slf4j.LoggerFactory;
-
 public class SmallMazeGenerator extends MazeGenerator {
 
     @Override
     public void build() {
-        setRadius(calculateDefaultMazeSize() * 0.5f);
+        setRadius(calculateDefaultMazeSize() * 0.6f);
```
Comment (IsaacLic) [inline engine/src/main/java/org/destinationsol/world/generators/SmallMazeGenerator.java:22]: Just confirming this was intentional - resolve this if it was

**Review by IsaacLic — COMMENTED** [review]
Looks good to me! If someone can test it and confirm it works well, I'll merge it

**Comment by Cervator** [comment]
[Finalise and merge GSoC 2021](https://trello.com/c/difqGsTb/2-finalise-and-merge-gsoc-2021)

**Comment by Cervator** [comment]
I've tested this locally although my Android setup might be broken - I am pretty sure I saw a new maze and everything seemed to work. Even with a few outstanding comments I'm going to go ahead and merge this and make a follow-up issue :-)

**Pull request merged**